### DISCUSSION TITLE
Support wider range of Python versions?

### DISCUSSION THREAD

**Discussion by jgarte** [Abjad/abjad#1493] [Q&A]
@trevorbaca 

Would it be possible for `abjad` to support a wider range of Python versions instead of just the bleeding edge one?

WDYT

**Comment by trevorbaca** [comment]
Why?

**Reply by jgarte** [reply]
One reason is that I haven't been able to update the abjad package in GNU Guix because abjad moves quicker than the Python versions in GNU Guix's main channel. Guix has had Python 3.9.9 in it's master branch (primary channel that most people use) for the last couple of abjad releases that were greater than or equal to 3.10.

The other remedy is for Guix people to package Python versions in the primary channel faster but I think there will always be a timeframe where the latest abjad version will be too new to be in the main Guix channel and we'll have to wait to upgrade Python which might take up to or more than 6 months since all dependent packages have to be rebuilt gradually without breakage.

**Comment by trevorbaca** [comment]
The trick is that the Abjad releases that require Python 3.10 use some of the 3.10-only functionality introduced in that release: type-union aliases (PEP 604) and the `typing.TypeAlias` class (PEP 613), and maybe a few others.
